app/src/main/java/com/example/footprnt/Discover/Adapters/ListAdapter.java

```diff
@@ -136,7 +147,28 @@ public boolean onTouch(View v, MotionEvent event) {
                     return true;
                 }
             });
-            tvBusinessPhone.setText(business.getPhone());
+            }
+            final String businessPhoneNum = business.getPhone();
+            if (businessPhoneNum != null && businessPhoneNum.length() > 0){
+                tvBusinessPhone.setVisibility(View.VISIBLE);
+                tvBusinessPhone.setText(businessPhoneNum);
+                tvBusinessPhone.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        Intent intent = new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + businessPhoneNum));
+                        ((Activity) mContext).startActivityForResult(intent, 20);
+                    }
+                });
+                btnCall.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        Intent intent = new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + businessPhoneNum));
+                        ((Activity) mContext).startActivityForResult(intent, 20);
```

Comment:
20 should be constant so that there's less chance of people using the same value (because it's very obviously at the top of the file and named REQUEST_CODE_XXX) and so that it doesn't get out of sync with the onActivityResult